lib/collection/src/shards/shard_holder/mod.rs

```diff
@@ -533,6 +526,16 @@ impl ShardHolder {
             .unwrap_or_default()
         {
             ShardingMethod::Auto => {
+                let mut shard_number = shard_number;
+
+                // If resharding is active, we should load an extra shard
+                // TODO(resharding): prevent error if extra shard cannot be loaded?
+                // TODO(resharding): may be needed if resharding starts and crashes before migrate
+                let is_resharding = self.resharding_state.read().is_some();
+                if is_resharding {
+                    shard_number += 1;
+                }
```

Comment:
I'm not 100% sure about this one. Should we, maybe, increment `shard_number` at the point when we *create* resharding shard (i.e., when we start resharding)?

And add code to decrement it if we abort.

Seems easier this way, I think.

Reply:
Yes, that's probably better.

On abort, we also need to explicitly remove that extra shard. That's probably a better place to decrease the count.

Reply:
a47baaa095e8f670e0d184e95a0877e193b1f7c9